### PR TITLE
CR-1135808 Some improvement suggestions for throttling feature

### PR DESCRIPTION
**Pull request by rchane**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1135808 Output for user could use clarification for cmc report. "Scaling" should be renamed to "throttling" for SubCmdConfigure.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Implemented suggestions 1 and 2. Suggestions 3 and 4 were taken care of previously.
#### How problem was solved, alternative solutions (if any) and why they were rejected
xbutil no longer outputs cmc status, as one should use xbmgmt to check and enable/disable CTS (clock throttling/shutdown). xbmgmt will advise users to double check if an xclbin is loaded when checking throttling status. A ptree change to check xclbin id was considered but was deemed insufficient. 
#### Risks (if any) associated the changes in the commit
N/A
#### What has been tested and how, request additional testing if necessary
xbutil will not show cmc status.
```
rchane@xsjsagarw50:/proj/rdi/staff/rchane/XRT$ xbutil examine -d 17:00 -r cmc
XRT build version: 2.15.0
Build hash: 3445fc127610829976a7da2f43488d4e2c6e1de9
Build date: 2023-02-06 16:27:56
Git branch: CR-1135808
PID: 319994
UID: 90979
[Tue Feb  7 00:34:24 2023 GMT]
HOST: xsjsagarw50
EXE: /proj/xsjhdstaff4/rchane/XRT/build/Debug/opt/xilinx/xrt/bin/unwrapped/xbutil2
[xbutil] ERROR: No report generator found for report: 'cmc'
: Invalid argument
```
Runtime Clock Scaling is now Clock Throttling and Shutdown:
```
rchane@xsjsagarw50:/proj/rdi/staff/rchane/XRT$ xbmgmt examine -d 17:00 -r cmc

------------------------------------------
[0000:17:00.0] : xilinx_u30_gen3x4_base_2
------------------------------------------
CMC
  Heartbeat information unavailable
  Clock Throttling and Shutdown:
    Not supported
```
Clock Throttling and Shutdown status now warns the user in the "not supported" case:
```
  Clock Throttling and Shutdown:
    Not supported. Ensure xclbin is loaded.
```
Renamed scaling to throttling when showing device configuration:
```
0000:17:00.0
  Security level                   : 0
  Clock Throttling enabled         : Not supported
  Throttling threshold power override: Not supported
  Throttling threshold temp override: Not supported
  Data retention                   : disabled
```
Renamed scaling to throttling when displaying clock options:
```
  --clk_throttle     - Enable/disable the device clock throttling
  --ct_threshold_power_override - Update the power threshold in watts
  --ct_threshold_temp_override - Update the temperature threshold in celsius
  --ct_reset         - Reset all throttling options
```
#### Documentation impact (if any)
The github.io page should reflect the changes to the cmc report. https://jira.xilinx.com/browse/CR-1150767 will include documentation update to describe how to enable/disable CTS.

### CODE REVIEW

**Comment by chvamshi-xilinx** [comment]
Hi @rchane , Overall, changes are looking good, Can you please compare the json output with and without your changes.
There are more changes required to rename all the ini options from "scaling" to "throttling" in subCmdConfigure.cpp. Please make those changes in separate PR

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp [hunk 1]
```diff
@@ -78,12 +83,18 @@ ReportCmcStatus::getPropertyTree20202( const xrt_core::device * _pDevice,
...
     }
   }
   catch(const xrt_core::query::no_such_key&) {}
   catch(const xrt_core::query::sysfs_error&) {}
 
+  xrt::device device(_pDevice->get_device_id());
+  std::stringstream ss;
+  ss << device.get_info<xrt::info::device::dynamic_regions>();
+  boost::property_tree::read_json(ss, device_stat_tree);
+  cmc_tree.add_child("device_stats", device_stat_tree);
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/common/ReportCmcStatus.cpp:96]: We definitely should not be placing the entire dynamic region report into the cmc property tree.

If we want the xclbin uuid there are better ways of getting it other than going the dynamic region report.
```suggestion
  cmc_tree.add_child("xclbin_uuid", _pDevice->get_xclbin_uuid());
```
We could also just state if the xclbin is valid through the property tree
```suggestion
  cmc_tree.add_child("cmc_status_valid", boost::iequals(_pDevice->get_xclbin_uuid(), N/A"));
```

**Review by dbenusov-xilinx — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp [hunk 2]
```diff
@@ -52,8 +53,8 @@ ReportCmcStatus::getPropertyTree20202( const xrt_core::device * _pDevice,
   catch(const xrt_core::query::sysfs_error&) {}
 
   try {
-    runtime_tree.put("Description", "Runtime Clock Scaling");
-	auto clk_scaling_data = xrt_core::device_query<xrt_core::query::clk_scaling_info>(_pDevice);
+    runtime_tree.put("Description", "Clock Throttling and Shutdown");
+	  auto clk_scaling_data = xrt_core::device_query<xrt_core::query::clk_scaling_info>(_pDevice);
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/common/ReportCmcStatus.cpp:57]: Weird spacing here?

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp [hunk 3]
```diff
@@ -115,34 +116,35 @@ ReportCmcStatus::writeReport( const xrt_core::device* /*_pDevice*/,
...
-      _output << "    Not supported\n";
+    boost::property_tree::ptree cmc_throttle = cmc.get_child("throttling");
+    _output << boost::format("  %-22s:\n") % cmc_throttle.get<std::string>("Description");
+
+    if (!cmc_throttle.get<bool>("supported")) {
+      _output << "    Not supported. Ensure xclbin is loaded.\n";
       return;
     }
-    if (!cmc_scale.get<bool>("enabled")) {
+    if (!cmc_throttle.get<bool>("enabled")) {
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/common/ReportCmcStatus.cpp:126]: Add a new line to separate the if statements. I thought they were an if/elseif grouping!

**Review by dbenusov-xilinx — APPROVED** [review]
Minor changes, nothing to stop merging though. Looks great!

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by chvamshi-xilinx** [comment]
Hi @rchane , Please fix formatting issues raised by Daniel in next PR.

**Pull request merged**